### PR TITLE
sql-statements: update the compability comment for `SHOW INDEX`

### PR DESCRIPTION
**Pull request by winoros**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Update the comments for `SHOW INDEX` after the pull [tidb#42441](https://github.com/pingcap/tidb/pull/42441)

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-show-indexes.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ SHOW KEYS FROM t1;
 
 ## MySQL 兼容性
 
-MySQL 中的 `Cardinality` 列返回该索引上不同值的个数，而 TiDB 中的 `Cardinality` 列始终返回 `0`。
+`SHOW INDEXES [FROM|IN]` 语句与 MySQL 完全兼容。如发现任何兼容性差异，请在 GitHub 上提交 [issue](https://github.com/pingcap/tidb/issues/new/choose)。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-show-indexes.md:101]: @winoros `STATISTICS` 表的字段描述是否需要修改

https://github.com/pingcap/docs-cn/blob/a9a57be13b9d051940a4a2da969cbd64a433b77c/information-schema/information-schema-statistics.md?plain=1#L55

**Review by Oreoxmt — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@winoros Could you please invite a tech reviewer? Thanks!

**Review by AilinKid — APPROVED** [review]

File: sql-statements/sql-statement-show-indexes.md (hunk 1 above)
Comment (Oreoxmt) [reply sql-statements/sql-statement-show-indexes.md:101]: https://github.com/pingcap/tidb/pull/42441 不影响 `STATISTICS` 表中 `CARDINALITY` 的行为

**Pull request merged**